jupyterhub/templates/scheduling/user-placeholder/statefulset.yaml

```diff
@@ -47,4 +47,6 @@ spec:
           image: {{ .Values.prePuller.pause.image.name }}:{{ .Values.prePuller.pause.image.tag }}
           resources:
             {{- include "jupyterhub.resources" . | nindent 12 }}
+      securityContext:
+        runAsUser: 1000
```

Comment:
This is a podSecurityContext, which can influence sidecar containers for example from `istio` which I've seen can lead to issues before, so I think we should set it on the container level only influencing the containers we actually manage.

Reply:
Yes, this does make sense. Adding it as a `podSecurityContext` seems better than for the entire deployment. Maybe we can add the `runAsGroup` parameter as well to ensure that any changes to the image does not affect us in the future?

The indentation in the code above seems to miss a tab, I think it should be as follows:

```
  image: {{ .Values.prePuller.pause.image.name }}:{{ .Values.prePuller.pause.image.tag }}
  resources:
    {{- include "jupyterhub.resources" . | nindent 12 }}
  securityContext:
    runAsGroup: 1000
    runAsUser: 1000
```

Reply:
Thanks for thinking about this @NerdSec ! I've made it a container securityContext in #1798, and also added runAsGroup there, what do you think?